wdl-lint/tests/lints/nonmatching-output/source.errors

```diff
@@ -71,26 +71,44 @@ warning[NonmatchingOutput]: output `v` is missing from `meta.outputs` section in
    │
    = fix: add a description of output `v` to documentation in `meta.outputs`
 
+note[MisplacedLintDirective]: lint directive `NonmatchingOutput` has no effect above: metadata object item
```

Comment:
> I think the comments relating to this line in the source file are now out of date.

I didn't look through the `source.wdl` files very closely, as I'm planning on rewriting many of them in a follow-up PR. If it's ok with you, I'm fine with leaving this as is for now.

> Should we not support the lint at the item level, though? Otherwise it's not possible to ignore particular names in the output section.

I'm not sure the right call here. My initial thinking was that adding metadata items potentially complicates the documentation. Take this example:
```
task {
    meta {
        #@ except: NonmatchingOutputs
        description: "this meta item is being excepted despite not being involved with the `NonmatchingOutputs` rule"
        outputs: {
            t: "t",
            s: "s",
        }
    }
    command <<< >>>
    output {
        String s = "hello"
        String t = "world"
    }
}
```
What diagnostic should be printed there? Should it just be ignored?

I also think the `NonmatchingOutputs` rule is appropriate to have as an "all or nothing rule". It compares ordering between different fields and I don't think there's an "obvious" solution for how to handle exceptions in the middle of one field but not in the other.

Reply:
Your concerns make sense to me. Let's leave it as-is and change accordingly if we receive user feedback in the future.